op-node/node/client.go

```diff
@@ -135,16 +141,48 @@ type L1EndpointConfig struct {
 	// L1RPCKind identifies the RPC provider kind that serves the RPC,
 	// to inform the optimal usage of the RPC for transaction receipts fetching.
 	L1RPCKind sources.RPCProviderKind
+
+	// RateLimit specifies a self-imposed rate-limit on L1 requests. 0 is no rate-limit.
+	RateLimit float64
+
+	// BatchSize specifies the maximum batch-size, which also applies as L1 rate-limit burst amount (if set).
+	BatchSize int
+
+	// HttpPollInterval specifies the interval between polling for the latest L1 block,
+	// when the RPC is detected to be an HTTP type.
+	// It is recommended to use websockets or IPC for efficient following of the changing block.
+	// Setting this to 0 disables polling.
+	HttpPollInterval time.Duration
 }
 
 var _ L1EndpointSetup = (*L1EndpointConfig)(nil)
 
-func (cfg *L1EndpointConfig) Setup(ctx context.Context, log log.Logger) (cl client.RPC, trust bool, kind sources.RPCProviderKind, err error) {
-	l1Node, err := client.NewRPC(ctx, log, cfg.L1NodeAddr)
+func (cfg *L1EndpointConfig) Check() error {
```

Comment:
Does it make sense to have a difference between L1 rpc config and L2 rpc config? Also does it make sense to have a difference between an L1 rpc client and an L2 rpc client? Feels like a single, robust client could be implemented and then naming of struct members could differentiate which network the client is dialed to

Reply:
The client code is already largely shared, it's a tree structure like this:
- eth-client -> L1-client
- eth-client -> L2-client
- L2-client -> engine client

Since each client adds additional more domain-specific caches / methods. The bulk of it is shared in the eth-client though.

The L1 configuration is a bit more separate, but since the L2 engine (JWT authenticated) and L1 (non-local, more important to optimize with rate-limits and polling etc. adjustments) are quite different, I don't think we should combine them here.